### DISCUSSION TITLE
Encountered a `504` - `request timed out` error while performing `auth` tasks (e.g., sending recovery or magic links) in a self-hosted Supabase Studio

### DISCUSSION THREAD

**Discussion by rafadlis** [supabase/supabase#30152] [Questions]
## Specs

| Spec | detail |
| --- | --- |
| OS | Ubuntu Server |
| Main Tools | aapanel |
| Supabase Deployment | Docker |
| Mail Server | Mail Server aapanel (tested) |

## SMTP config

the SMTP server is running fine but it takes quite a while to send an email.
 
## the logs

Event Message

```json
{
  "auth_event": {
    "action": "user_recovery_requested",
    "actor_id": "03016027-3423-4c7e-b262-6a8cac8e8594",
    "actor_username": "rahmatfadlis@gmail.com",
    "actor_via_sso": false,
    "log_type": "user"
  },
  "component": "api",
  "error": "context deadline exceeded",
  "level": "error",
  "method": "POST",
  "msg": "504: Processing this request timed out, please retry after a moment.",
  "path": "/recover",
  "referer": "http://localhost:3000",
  "remote_addr": "172.19.0.12",
  "request_id": "e180d34f-af64-4e10-acb5-f6fdf2d773bb",
  "time": "2024-10-28T22:47:16Z"
}
```

metadata

```json
[
  {
    "auth_event": [
      {
        "action": "user_recovery_requested",
        "actor_id": "03016027-3423-4c7e-b262-6a8cac8e8594",
        "actor_username": "rahmatfadlis@gmail.com",
        "actor_via_sso": false,
        "log_type": "user"
      }
    ],
    "component": "api",
    "error": "context deadline exceeded",
    "level": "error",
    "method": "POST",
    "msg": "504: Processing this request timed out, please retry after a moment.",
    "path": "/recover",
    "referer": "http://localhost:3000",
    "remote_addr": "172.19.0.12",
    "request_id": "e180d34f-af64-4e10-acb5-f6fdf2d773bb",
    "time": "2024-10-28T22:47:16Z",
    "timestamp": "2024-10-28T22:47:16Z"
  }
]
```

## The Questions

How can I extend the request timeout duration? Are there any alternative solutions to address this issue?

**Comment by abdrhxyii** [comment]
Did you fix the issue ?. Im also having the same issue.

**Reply by rafadlis** [reply]
My proposed solution is to switch the mail server to resend.com and use different domain, configure it using `.env` settings as outlined in Resend's documentation:

[resend.com/docs/send-with-supabase-smtp](https://resend.com/docs/send-with-supabase-smtp)

However, I haven't been able to find a solution and don't have enough time to debug it.

**Reply by rafadlis** [reply]
My biggest hypothesis is that there might be an issue with the `.env` configuration I set or the domain I used

**Reply by carbonkel525** [reply]
If you using resend, keep the configuration as in the docs. Only put your port to 587. It worked for me.

**Reply by rafadlis** [reply]
> If you using resend, keep the configuration as in the docs. Only put your port to 587. It worked for me.

I will try this, thanks for the answer

**Reply by carbonkel525** [reply]
Let me know if it worked

**Reply by yerffejytnac** [reply]
587 worked for me

**Comment by carbonkel525** [comment]
Having the same problem here, using a self-hosted supabase instance via coolify. I set all the .env variables in the coolify dashboard for using the resend smtp server. `SMTP_HOST=smtp.resend.com`, `SMTP_PASS=my_resend_api_key`, `SMTP_PORT=465`, `SMTP_USER=resend`

When inviting a user in the auth>Users page in my self-hosted supabase instance, I get a 504 error in the console with message: Processing this request timed out, please retry after a moment.

**Reply by abdrhxyii** [reply]
I'm having this for 1 week, couldn't fix it yet.

**Reply by carbonkel525** [reply]
I tried the default coolify smtp port `SMTP_PORT=587` and it solved my issue. I don't know why resend is telling us to use `SMTP_PORT=465`.
Invite email and password recovery works great now.

**Reply by carbonkel525** [reply]
To make it more clear `SMTP_HOST=smtp.resend.com`, `SMTP_PASS=my_resend_api_key`, `SMTP_PORT=587`, `SMTP_USER=resend`, `ENABLE_EMAIL_AUTOCONFIRM=true`, `ENABLE_EMAIL_SIGNUP=true` try to set your .env vars like this. Then go to your supabase instance and try to invite a user. It should work

**Reply by rafadlis** [reply]
> I'm having this for 1 week, couldn't fix it yet.

If the @carbonkel525 solution isn't working, try changing the domain. It might help. Also, check the domain configuration and health for any issues.

**Comment by kyleorin** [comment]
having the same issue with zeptomail not entirely sure if im doing something wrong or its simply not working

**Comment by ewann** [comment]
fwiw: I just encountered similar / same situation, that was confirmed not the case as recently as yesterday. Below configuration resolved my incident:

<img width="870" alt="Screenshot 2025-01-20 at 13 19 16" src="https://github.com/user-attachments/assets/58ff2c47-64ba-46fe-b425-bdcd5a018d8a" />

Where as 465 had been configured for ~6 weeks previous without being an issue.

https://resend.com/docs/send-with-smtp captures the info:
<img width="742" alt="Screenshot 2025-01-20 at 13 21 26" src="https://github.com/user-attachments/assets/a74578e1-69b9-4d35-87bf-294e5d4590e1" />

and https://resend.com/docs/send-with-supabase-smtp still instructs to use 465. Looking at the previous comments in this discussion,  I suppose there is some rolling infrastructure change on supabase side that is getting deployed across all envs? I'm in eu-west-2 / London; perhaps interestingly there is currently a maintenance window advertised, although it allegedly doesn't impact my region
<img width="871" alt="Screenshot 2025-01-20 at 13 24 23" src="https://github.com/user-attachments/assets/1fef1424-4328-44c9-9f32-0cfaf64f905d" />


🤷‍♂️

**Comment by askarov-web2cat** [comment]
У меня была такая ошибка. Оказалось что мой сервис VPS блокировал почтовые порта, 587, 465, 25 и т.д. Я написал в поддержку и они открыли порты которые мне нужны были

**Comment by russoue** [comment]
Sharing in case it helps someone.  I was trying to use Amazon SES and in the beginning tried to use their STARTTLS port 25 and got the same error.  However, after switching to their TLS Wrapper Port 465 the errors went away and it seemed like the emails were being sent.  However, my Amazon SES account is still in sandbox mode and I haven't received the emails I sent to myself yet.  I am guessing that has something to do with the SES account and not GoTrue.

**Comment by jernland0279** [comment]
Having the same issue with cpanel host, the email server is hosted in cpanel and before it was working correcting until it suddenly stopped and started giving this issue. I tried to deactivate the smtps settings in supabase and it worked  but i would really want to use a custom smtps. From now, i don't kmow how to solve it.
